R/survfit2.R

```diff
@@ -91,5 +91,29 @@ survfit2 <- function(formula, ...) {
   # update object with env and add another class -------------------------------
   survfit %>%
     utils::modifyList(val = list(.Environment = env)) %>%
-    structure(class = c("survfit2", class(survfit)))
+    structure(class = c("survfit2", class(survfit))) %>%
+    .check_PARAM_consistency()
+}
+
+.check_PARAM_consistency <- function(x) {
+  # get the formula and data
+  formula <- .extract_formula_from_survfit(x)
+  data <- .extract_data_from_survfit(x)
+
+  if (is.null(data) || is.null(formula)) return(x)
+  if (all(c("PARAM", "PARAMCD") %in% names(data)) && !.is_PARAM_consistent(formula, data))
+    cli::cli_inform(c("!" = "Columns {.cls {c('PARAM', 'PARAMCD')}} are not unique and usage is likely incorrect."))
+
+  x
+}
+
+.is_PARAM_consistent <- function(formula, data) {
```

Comment:
One edge case: if only one param is present and user wants to use Surv_CNSR() ~ PARAMCD, data = adtte. This edge case occurs in preparation of a data base lock where multiple parameters are expected but currently only one can be calculated. Programmers prepare their programs upfront so when database lock comes they dont have to adjust anything anymore.

Reply:
Can we include this edge case in testing as well before we forget to add this as a requirement.

Reply:
Updated with checks in the scenario where only PARAMCD is present. unit tests included as well